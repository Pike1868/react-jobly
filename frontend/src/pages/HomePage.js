import React from "react";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";
import Box from "@mui/material/Box";

const HomePage = () => {
  return (
    <Container component="main" maxWidth="md">
      <Box sx={{ mt: 8, mb: 4 }}>
        <Typography variant="h2" component="h1" gutterBottom>
          Jobly
        </Typography>
        <Typography variant="h5" component="p" gutterBottom>
          All the jobs in one, convenient place.
        </Typography>
        <Typography variant="h4" component="h2">
          Welcome Back, [Username]!
        </Typography>
      </Box>
    </Container>
  );
};

export default HomePage;
