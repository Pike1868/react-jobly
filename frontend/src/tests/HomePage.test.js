import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import HomePage from "../pages/HomePage";
import { mockUserDetails } from "./commonSetup";

jest.mock("../context/UserContext", () => ({
  useUserContext: () => mockUserDetails,
}));

describe("Homepage", () => {
  it("renders without crashing", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    expect(asFragment).toMatchSnapshot();
  });
});
