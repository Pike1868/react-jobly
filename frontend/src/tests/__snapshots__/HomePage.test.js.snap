// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot 1`] = `[Function]`;
